### PR TITLE
Avoid invalid escape sequences, use raw strings

### PR DESCRIPTION
**Pull request by TurtleOrangina**
# What does this PR do?
Fixes invalid escape sequences in strings, they are illegal in python, and will throw a SyntaxError if run with "-W error", and always throw a syntax error starting with python-3.12 (planned). With python between 3.6 and 3.11 and "-W default" they produce a DeprecationWarning:
```
> python -W error -c '"(.*?)-\d{5}-of-\d{5}"'                                                                                                                                 ✔    File "<string>", line 1
    "(.*?)-\d{5}-of-\d{5}"
    ^^^^^^^^^^^^^^^^^^^^^^
SyntaxError: invalid escape sequence '\d'
```
This has been fixed in the past, e.g. https://github.com/huggingface/transformers/pull/4924 - but missing linter support and the fact that python only fails this with "-W" flag set has let the issue be re-introduced. This PR fixes those occurrences and enables ruff "W605" error, which will prevent this for the future. 

## Who can review?

Maybe
@sgugger who added ruff in the first place or (Why was "W605" disabled when switching to ruff?)
@patrickvonplaten wo recently introduced some invalid escape sequences
@LysandreJik who merged the linked fix from 2020

### CODE REVIEW

File: examples/research_projects/quantization-qdqbert/quant_trainer.py [hunk 1]
```diff
@@ -41,8 +41,10 @@ def add_arguments(parser):
     group.add_argument("--quant-disable", action="store_true", help="disable all quantizers")
     group.add_argument("--quant-disable-embeddings", action="store_true", help="disable all embeddings quantizers")
     group.add_argument("--quant-disable-keyword", type=str, nargs="+", help="disable quantizers by keyword")
-    group.add_argument("--quant-disable-layer-module", type=str, help="disable quantizers by keyword under layer.\d+.")
-    group.add_argument("--quant-enable-layer-module", type=str, help="enable quantizers by keyword under layer.\d+.")
+    group.add_argument(
+        "--quant-disable-layer-module", type=str, help=r"disable quantizers by keyword under layer.\d+."
+    )
+    group.add_argument("--quant-enable-layer-module", type=str, help=r"enable quantizers by keyword under layer.\d+.")
```
Comment (sgugger) [inline examples/research_projects/quantization-qdqbert/quant_trainer.py:47]: Those changes don't make any sense. This is docstring, there shouldn't be no `\d+` at the end and no r.

File: src/transformers/models/maskformer/modeling_maskformer.py [hunk 2]
```diff
@@ -1250,7 +1250,7 @@ def __init__(self, *args, feature_size: int = 256, mask_feature_size: int = 256,
             feature_size (`int`, *optional*, defaults to 256):
                 The feature size (channel dimension) of the FPN feature maps.
             mask_feature_size (`int`, *optional*, defaults to 256):
-                The features (channels) of the target masks size \\C_{\epsilon}\\ in the paper.
+                The features (channels) of the target masks size \C_{\epsilon}\ in the paper.
```
Comment (sgugger) [inline src/transformers/models/maskformer/modeling_maskformer.py:1253]:
```suggestion
                The features (channels) of the target masks size \\(C_{\epsilon}\\) in the paper.
```
No here a parenthesis was missing only for the synatx for LaTeX rendering.

**Review by sgugger — COMMENTED** [review]
Thanks for fixing those. I have pointed the ones that shouldn't be fixed this way in comments.

File: src/transformers/models/maskformer/modeling_maskformer.py (hunk 2 above)
Comment (Lingepumpe) [reply src/transformers/models/maskformer/modeling_maskformer.py:1253]: added as suggested

**Review by sgugger — APPROVED** [review]
Thanks a lot!

**Comment by sgugger** [comment]
Failures are unrelated to this PR and due to the last release of huggingface_hub. All is fixed on main so merging :-)

**Comment by Lingepumpe** [comment]
> Rebased on top of current main branch to pass the tests.

Does not seem to help the tests, let me know if I should do anything else for this PR

**Review by helpmefindaname — APPROVED** [review]

**Comment by sgugger** [comment]
The Hub is currently having high-response times due to some abusive traffic, which is why the tests are all red. I just forgot to push the merge button yesterday, so merging this as it shouldn't have any negative impact on main.

**Pull request merged**